Revert Value Reading and Add Read Method to GamepadEx

JIceberg opened the pull request:
# Efficient Gamepad

Please note that we accept pull requests from anyone, but that does not mean it will be merged.

## What kind of change does this PR introduce?
* Feature - makes the use of readers more efficient to use only one value read per loop (reverts the change made in #109) and adds a single `readButtons()` method to `GamepadEx` to read all button values at once

## Did this PR introduce a breaking change?
* No

__Please make sure your PR satisfies the requirements of [the contributing page](CONTRIBUTING.md)__

puravdatta-sudo reviewed (approved):
Looks good.

litehed reviewed (approved):
Looks good, but I was just wondering what makes this more efficient or better to what was overwritten in the change? Just Curious.

JIceberg commented:
> Looks good, but I was just wondering what makes this more efficient or better to what was overwritten in the change? Just Curious.

So, the reason we only want to call `readValue()` once is because the gamepad values are updated on a separate thread. This means that it can change several times in one loop. So if I have
```java
boolean x = gamepad1.x;
...
boolean y = gamepad1.x;
```
Then it's possible that x != y is true, which could lead to problems in the control logic. So we only read the value once and use that one value for all of our logic. The reason why the change in PR #109  was made previously was because button readers were a massive pain. If you had several, you'd end up with just something like this:
```java
while (...) {
  ...
  xReader.readValue();
  yReader.readValue();
  aReader.readValue();
  bReader.readValue();
  lBumperReader.readValue();
  ...
}
```
which is a bit of a pain to type out and rather ugly. But the PR also added a map of button reader objects to `GamepadEx`. So we can actually circumvent that with a single call that does the exact same thing:
```java
gamepadEx.readButtons();
```
Which calls `readValue()` on every button reader in the map. So now we've simplified it from several `readValue()` calls to just one method call and no extra objects.
